### ISSUE TITLE
After the program runs to this point, it will not continue<Using /home/zz/.cache/torch_extensions as PyTorch extensions root... >

### ISSUE THREAD

**Issue by ZbearM** [lucidrains/DALLE-pytorch#172]
After I use the distributed training running program of deepspeed, the program runs to < using / home / zz /. cache / torch_ extensions as pytorch extensions root... >  will stop at this step and no longer run. Is there any solution?

**Comment by ZbearM** [comment]
``[2021-04-07 08:57:31,293] [INFO] [logging.py:60:log_dist] [Rank 0] DeepSpeed info: version=0.3.13, git-hash=unknown, git-branch=unknown
[2021-04-07 08:57:31,300] [INFO] [engine.py:75:_initialize_parameter_parallel_groups] data_parallel_size: 3, parameter_parallel_size: 3
[2021-04-07 08:57:32,666] [INFO] [engine.py:75:_initialize_parameter_parallel_groups] data_parallel_size: 3, parameter_parallel_size: 3
[2021-04-07 08:57:33,157] [INFO] [engine.py:75:_initialize_parameter_parallel_groups] data_parallel_size: 3, parameter_parallel_size: 3
[2021-04-07 08:57:33,292] [INFO] [engine.py:591:_configure_optimizer] Removing param_group that has no 'params'in the client Optimizer
[2021-04-07 08:57:33,298] [INFO] [engine.py:591:_configure_optimizer] Removing param_group that has no 'params'in the client Optimizer
[2021-04-07 08:57:33,299] [INFO] [engine.py:591:_configure_optimizer] Removing param_group that has no 'params'in the client Optimizer
[2021-04-07 08:57:33,299] [INFO] [engine.py:596:_configure_optimizer] Using client Optimizer as basic optimizer
[2021-04-07 08:57:33,299] [INFO] [engine.py:605:_configure_optimizer] DeepSpeed Basic Optimizer = Adam
[2021-04-07 08:57:33,300] [INFO] [logging.py:60:log_dist] [Rank 0] DeepSpeed Final Optimizer = Adam
[2021-04-07 08:57:33,300] [INFO] [engine.py:443:_configure_lr_scheduler] DeepSpeed using client LR scheduler
[2021-04-07 08:57:33,301] [INFO] [logging.py:60:log_dist] [Rank 0] DeepSpeed LR Scheduler = None
[2021-04-07 08:57:33,302] [INFO] [logging.py:60:log_dist] [Rank 0] step=0, skipped=0, lr=[0.0003], mom=[(0.9, 0.999)]
[2021-04-07 08:57:33,303] [INFO] [config.py:737:print] DeepSpeedEngine configuration:
[2021-04-07 08:57:33,303] [INFO] [config.py:741:print]   activation_checkpointing_config  {
    "contiguous_memory_optimization": false,
    "cpu_checkpointing": false,
    "number_checkpoints": null,
    "partition_activations": false,
    "profile": false,
    "synchronize_checkpoint_boundary": false
}
[2021-04-07 08:57:33,305] [INFO] [config.py:741:print]   allreduce_always_fp32 ........ False
[2021-04-07 08:57:33,305] [INFO] [config.py:741:print]   amp_enabled .................. False
[2021-04-07 08:57:33,305] [INFO] [config.py:741:print]   amp_params ................... False
[2021-04-07 08:57:33,309] [INFO] [config.py:741:print]   checkpoint_tag_validation_enabled  True
[2021-04-07 08:57:33,309] [INFO] [config.py:741:print]   checkpoint_tag_validation_fail  False
[2021-04-07 08:57:33,312] [INFO] [config.py:741:print]   disable_allgather ............ False
[2021-04-07 08:57:33,312] [INFO] [config.py:741:print]   dump_state ................... False
[2021-04-07 08:57:33,312] [INFO] [config.py:741:print]   dynamic_loss_scale_args ...... None
[2021-04-07 08:57:33,313] [INFO] [config.py:741:print]   elasticity_enabled ........... False
[2021-04-07 08:57:33,313] [INFO] [config.py:741:print]   flops_profiler_config ........ {
    "detailed": true,
    "enabled": false,
    "module_depth": -1,
    "profile_step": 1,
    "top_modules": 3
}
[2021-04-07 08:57:33,314] [INFO] [config.py:741:print]   fp16_enabled ................. False
[2021-04-07 08:57:33,314] [INFO] [config.py:741:print]   global_rank .................. 0
[2021-04-07 08:57:33,314] [INFO] [config.py:741:print]   gradient_accumulation_steps .. 1
[2021-04-07 08:57:33,317] [INFO] [config.py:741:print]   gradient_clipping ............ 0.0
[2021-04-07 08:57:33,317] [INFO] [config.py:741:print]   gradient_predivide_factor .... 1.0
[2021-04-07 08:57:33,318] [INFO] [config.py:741:print]   initial_dynamic_scale ........ 4294967296
[2021-04-07 08:57:33,318] [INFO] [config.py:741:print]   loss_scale ................... 0
[2021-04-07 08:57:33,318] [INFO] [config.py:741:print]   memory_breakdown ............. False
[2021-04-07 08:57:33,320] [INFO] [config.py:741:print]   optimizer_legacy_fusion ...... False
[2021-04-07 08:57:33,320] [INFO] [config.py:741:print]   optimizer_name ............... None
[2021-04-07 08:57:33,321] [INFO] [config.py:741:print]   optimizer_params ............. None
[2021-04-07 08:57:33,321] [INFO] [config.py:741:print]   pipeline ..................... {'stages': 'auto', 'partition': 'best', 'seed_layers': False, 'activation_checkpoint_interval': 0}
[2021-04-07 08:57:33,321] [INFO] [config.py:741:print]   pld_enabled .................. False
[2021-04-07 08:57:33,324] [INFO] [config.py:741:print]   pld_params ................... False
[2021-04-07 08:57:33,324] [INFO] [config.py:741:print]   prescale_gradients ........... False
[2021-04-07 08:57:33,324] [INFO] [config.py:741:print]   scheduler_name ............... None
[2021-04-07 08:57:33,325] [INFO] [config.py:741:print]   scheduler_params ............. None
[2021-04-07 08:57:33,326] [INFO] [config.py:741:print]   sparse_attention ............. None
[2021-04-07 08:57:33,326] [INFO] [config.py:741:print]   sparse_gradients_enabled ..... False
[2021-04-07 08:57:33,326] [INFO] [config.py:741:print]   steps_per_print .............. 10
[2021-04-07 08:57:33,328] [INFO] [config.py:741:print]   tensorboard_enabled .......... False
[2021-04-07 08:57:33,328] [INFO] [config.py:741:print]   tensorboard_job_name ......... DeepSpeedJobName
[2021-04-07 08:57:33,328] [INFO] [config.py:741:print]   tensorboard_output_path ...... 
[2021-04-07 08:57:33,328] [INFO] [config.py:741:print]   train_batch_size ............. 12
[2021-04-07 08:57:33,329] [INFO] [config.py:741:print]   train_micro_batch_size_per_gpu  4
[2021-04-07 08:57:33,329] [INFO] [config.py:741:print]   wall_clock_breakdown ......... False
[2021-04-07 08:57:33,330] [INFO] [config.py:741:print]   world_size ................... 3
[2021-04-07 08:57:33,330] [INFO] [config.py:741:print]   zero_allow_untested_optimizer  False
[2021-04-07 08:57:33,332] [INFO] [config.py:741:print]   zero_config .................. {
    "allgather_bucket_size": 500000000,
    "allgather_partitions": true,
    "contiguous_gradients": false,
    "cpu_offload": false,
    "cpu_offload_params": false,
    "cpu_offload_use_pin_memory": "cpu_offload_use_pin_memory",
    "elastic_checkpoint": true,
    "load_from_fp32_weights": true,
    "max_live_parameters": 1000000000,
    "max_reuse_distance": 1000000000,
    "overlap_comm": false,
    "param_persistence_threshold": 100000,
    "prefetch_bucket_size": 50000000,
    "reduce_bucket_size": 500000000,
    "reduce_scatter": true,
    "stage": 0,
    "sub_group_size": 1000000000000
}
[2021-04-07 08:57:33,333] [INFO] [config.py:741:print]   zero_enabled ................. False
[2021-04-07 08:57:33,333] [INFO] [config.py:741:print]   zero_optimization_stage ...... 0
[2021-04-07 08:57:33,333] [INFO] [config.py:743:print]   json = {
    "fp16":{
        "enabled":false
    },
    "train_batch_size":12
}
Using /home/zz/.cache/torch_extensions as PyTorch extensions root...
Using /home/zz/.cache/torch_extensions as PyTorch extensions root...
Using /home/zz/.cache/torch_extensions as PyTorch extensions root...

**Comment by afiaka87** [comment]
@ZbearM 
Are you running this on a custom setup? I ask because of the `zz` home directory. 

If so have you tried a run without deepspeed to start with? May help us narrow things down a bit.

**Comment by ZbearM** [comment]
Because I only have normal user rights on my account, I need to customize the directory to zz. Also, I can run normally on a GPU without the command of deepspeed, so I don't know why I can't run with deepspeed

**Comment by ZbearM** [comment]
> 
> 
> @ZbearM
> Are you running this on a custom setup? I ask because of the `zz` home directory.
> 
> If so have you tried a run without deepspeed to start with? May help us narrow things down a bit.

Because I only have normal user rights on my account, I need to customize the directory to ZZ. Also, I can run normally on a GPU without the command of deepspeed, so I don't know why I can't run with deepspeed

**Comment by janEbert** [comment]
I'm guessing DeepSpeed doesn't like your torch extension. It's not the best help but take a look at the [DeepSpeed issues (filtered by "torch_extensions"](https://github.com/microsoft/DeepSpeed/issues?q=is%3Aissue+torch_extensions+). There are several issues listed.

**Comment by ZbearM** [comment]
@afiaka87 lol..thanks a lot, Thank you. I'll go to the link you gave me to find the answer
